Content.Client/Instruments/MidiParser/MidiStreamWrapper.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System.IO;
+using System.Text;
+
+namespace Content.Client.Instruments.MidiParser;
+
+public sealed class MidiStreamWrapper
+{
+    private readonly MemoryStream _stream;
+    private byte[] _buffer;
+
+    public long StreamPosition => _stream.Position;
+
+    public MidiStreamWrapper(byte[] data)
+    {
+        _stream = new MemoryStream(data);
```

Comment:
Mark as read-only. Can't hurt.

```suggestion
        _stream = new MemoryStream(data, false);
```